sympy/functions/combinatorial/numbers.py

```diff
@@ -1861,9 +1861,29 @@ def stirling(n, k, d=None, kind=2, signed=False):
 def _nT(n, k):
     """Return the partitions of ``n`` items into ``k`` parts. This
     is used by ``nT`` for the case when ``n`` is an integer."""
+    if k > n or k < 0:
+        return 0
+    if k == n or k == 1:
+        return 1
     if k == 0:
-        return 1 if k == n else 0
-    return sum(_nT(n - k, j) for j in range(min(k, n - k) + 1))
+        return 0
+    if k == 2:
+        return n//2
+    d = n - k
+    if d <= 2:
+        return d
+    #The number of partitions of n into k parts is equal to the
+    #number of partitions of n-k, if k >= n - k so to increase
+    #the speed we are calling partition._partition(n - k).
+    if k >= d:
+        return partition._partition(d)
```

Comment:
Is this code never called? How are we getting by with passing 1 argument when the function takes 2?

Reply:
The caching mechanism provides the second argument.

Reply:
The function we are calling is actually a wrapper with one argument.